src/test/scala/org/apache/spark/sql/delta/DeltaSourceSuite.scala

```diff
@@ -991,6 +995,162 @@ class DeltaSourceSuite extends DeltaSourceSuiteBase {
       }
     }
   }
+
+  /** Generate commits with the given timestamp in millis. */
+  private def generateCommits(location: String, commits: Long*): Unit = {
+    val deltaLog = DeltaLog.forTable(spark, location)
+    var startVersion = deltaLog.snapshot.version + 1
+    commits.foreach { ts =>
+      val rangeStart = startVersion * 10
+      val rangeEnd = rangeStart + 10
+      spark.range(rangeStart, rangeEnd).write.format("delta").mode("append").save(location)
+      val file = new File(FileNames.deltaFile(deltaLog.logPath, startVersion).toUri)
+      file.setLastModified(ts)
+      startVersion += 1
+    }
+  }
+
+  private implicit def durationToLong(duration: FiniteDuration): Long = {
+    duration.toMillis
+  }
+
+  test("startingVersion and startingTimestamp") {
+    withTempDirs { (inputDir, outputDir, checkpointDir) =>
+      val tblLoc = inputDir.getCanonicalPath
+      val start = 1594418440842L
+      generateCommits(tblLoc, start, start + 20.minutes)
+
+      val q = spark.readStream
+        .format("delta")
+        .option("startingVersion", "1")
+        .load(inputDir.getCanonicalPath)
+        .writeStream
+        .format("delta")
+        .option("checkpointLocation", checkpointDir.getCanonicalPath)
+        .start(outputDir.getCanonicalPath)
+      try {
+        q.processAllAvailable()
+        checkAnswer(
+          spark.read.format("delta").load(outputDir.getAbsolutePath),
+          (0 until 20).map(_.toLong).toDF())
```

Comment:
why is this not 10 until 20?

Reply:
Just because it is a later version, wouldn't that version also contain all previously written data from version 0?

Reply:
we want to pull the incremental changes made to the table from that version onwards, so I'd expect this to start at 10, not include 0-10